### DISCUSSION TITLE
How to use a bibliography citation in a Latex table within Quarto?

### DISCUSSION THREAD

**Discussion by plierouge** [quarto-dev/quarto-cli#5924] [Q&A] [answered]
### Description

Hello,

I am new to Quarto. I know a little about Latex. I am writing a scientific paper with R code in Quarto (R Studio). I found that creating table in plain Markdown code is too limited. So I create a latex table. Example:

\begin{table}[]
\centering
\caption{Blah}
\label{tab:my-table}
\begin{tabular}{lll}
\hline
      & Title 1 & Title 2 \\
\hline
Row 1 & 12      & 14      \\
Row 2 & 11      & 16      \\
\hline
\multicolumn{3}{l}{}     
\end{tabular}
\end{table}

Let say I want to add a * after Title 2 and put a bibliographic reference in the { } of the \multicolumn line. 

What code should I use? I tried @{label} and \cite{label}. Nothing works.

Could someone help me?

Thanks!

**Comment by mcanouil** [comment]
Hi and welcome!

Use HTML table with markdown in it, Quarto will take care of the rest, see <https://quarto.org/docs/prerelease/1.3/tables.html#embedded-markdown-content-can-be-specified>.
Note you need Quarto >= 1.3.

**Reply by plierouge** [reply]
Hello!
Thank you for your comment. I will keep this in mind when I will be working in HTML. Unfortunately, the code does not work with pdf.

Do you have another recommendation?
Thanks!

**Reply by mcanouil** [reply]
What do you need it does not work?

Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by plierouge** [reply]
Hello!

Here's my latex table code:

```{=latex}
\begin{table}[]
\centering
\caption{Total amino acid composition of experimental diets (mass \% of TTA). See Table \ref{tbl-feed} for diet abbreviations.}
\label{tbl-aa}
\begin{tabular}{lrrrr}
\hline
\textbf{Amino acids}          & \textbf{FMD} & \textbf{CHD} & \textbf{HTD} & \textbf{Requirement for \textit{Salmo salar}}$^{a}$ \\
\hline
\textit{Essential (EAA)}      &              &              &              &            \\
Arginine                      & 7.2          & 7.1          & 9.4          & 4.1--5.1   \\
Histidine                     & 2.5          & 2.5          & 2.4          & 1.8--2.2   \\
Isoleucine                    & 4.4          & 4.4          & 4.2          & 3.2        \\
Leucine                       & 10.3         & 10.3         & 10.1         & 5.2        \\
Lysine                        & 3.7          & 3.7          & 3.4          & 3.2--6.1   \\
Methionine                    & 2.7          & 2.6          & 2.5          & 3.1        \\
Phenylalanine (Phe+Tyr)       & 4.8 (9.2)    & 4.8 (9.2)    & 4.7 (8.9)    & (5.8)      \\
Threonine                     & 4.1          & 4.1          & 4.0          & 3.2        \\
Tryptophan                    & 0.1          & 0.1          & 0.1          & --         \\
Valine                        & 5.4          & 5.4          & 5.2          & 3.9        \\
                              &              &              &              &            \\
\textit{Non-essential (NEAA)} &              &              &              &            \\
Alanine                       & 6.6          & 6.6          & 6.4          & --         \\
Aspartic acid                 & 8.9          & 8.9          & 8.4          & --         \\
Cystine                       & 1.3          & 1.4          & 1.3          & --         \\
Glutamic acid                 & 17.7         & 17.7         & 17.4         & --         \\
Glycine                       & 4.1          & 4.1          & 4.2          & --         \\
Hydroxyproline                & 0.3          & 0.3          & 0.2          & --         \\
Proline                       & 6.4          & 6.4          & 6.5          & --         \\
Serine                        & 4.6          & 4.6          & 4.6          & --         \\
Taurine                       & 0.3          & 0.4          & 0.6          & --         \\
Tyrosine                      & 4.4          & 4.4          & 4.2          & --         \\
\hline
\multicolumn{5}{l}{\tiny{TAA: Total amino acid}} \\[-1ex]
\multicolumn{5}{l}{\tiny{$^{a}$Lall and Anderson (2005)}} \\[-1ex]
\multicolumn{5}{l}{{\cite{lallAminoAcidNutrition2005}}}
\end{tabular}
\end{table}
```
Please look at the attached pdf output. At the end of the header, after Requirement for Salmo salar, I have $^{a}$  that I want to link to a bibliographic citation at the bottom of the table [ i.e. \multicolumn{5}{l}{{\cite{lallAminoAcidNutrition2005}}} ]. The result just show [?] in the output and not Lall and Anderson (2005).

Thanks for your help!

**Reply by mcanouil** [reply]
Thanks.
Could you use code block to format the code?
Also I meant the HTML table code you used that "did not work".

**Reply by plierouge** [reply]
Hello!

Is it this you suggest?

---
title: "html_table"
format: pdf
---

<style type="text/css">
.tg  {border-collapse:collapse;border-spacing:0;}
.tg td{border-style:solid;border-width:0px;font-family:Arial, sans-serif;font-size:14px;overflow:hidden;
  padding:10px 5px;word-break:normal;}
.tg th{border-style:solid;border-width:0px;font-family:Arial, sans-serif;font-size:14px;font-weight:normal;
  overflow:hidden;padding:10px 5px;word-break:normal;}
.tg .tg-0lax{text-align:left;vertical-align:top}
.tg .tg-tf2e{text-align:left;vertical-align:top}
</style>
<table class="tg">
<thead>
  <tr>
    <th class="tg-0lax">C1</th>
    <th class="tg-0lax">C2</th>
    <th class="tg-0lax">C3</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td class="tg-tf2e">1</td>
    <td class="tg-tf2e">2</td>
    <td class="tg-tf2e">3</td>
  </tr>
  <tr>
    <td class="tg-0lax">4</td>
    <td class="tg-0lax">5</td>
    <td class="tg-0lax">6</td>
  </tr>
</tbody>
</table>

My file name is html_table.qmd, so a Quarto file.

**Reply by plierouge** [reply]
![Capture d’écran du 2023-06-16 07-51-40](https://github.com/quarto-dev/quarto-cli/assets/136630150/027f06bd-b911-4f94-bcef-581837cd35bf)

**Reply by mcanouil** [reply]
Could use code blocks as shown in my example?
It's pretty hard to read un formatted code, also we lose the ability to make clean copy of the code for reuse/test purposes.

Also take a second read at <https://quarto.org/docs/prerelease/1.3/tables.html#embedded-markdown-content-can-be-specified>.

**Reply by plierouge** [reply]
OK sorry... first time using this forum!  A table generated by [https://www.tablesgenerator.com/](url)
````qmd
---
title: "html_table"
format: pdf
---

<style type="text/css">
.tg  {border-collapse:collapse;border-spacing:0;}
.tg td{border-style:solid;border-width:0px;font-family:Arial, sans-serif;font-size:14px;overflow:hidden;
  padding:10px 5px;word-break:normal;}
.tg th{border-style:solid;border-width:0px;font-family:Arial, sans-serif;font-size:14px;font-weight:normal;
  overflow:hidden;padding:10px 5px;word-break:normal;}
.tg .tg-0lax{text-align:left;vertical-align:top}
.tg .tg-tf2e{text-align:left;vertical-align:top}
</style>
<table class="tg">
<thead>
  <tr>
    <th class="tg-0lax">C1</th>
    <th class="tg-0lax">C2</th>
    <th class="tg-0lax">C3</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td class="tg-tf2e">1</td>
    <td class="tg-tf2e">2</td>
    <td class="tg-tf2e">3</td>
  </tr>
  <tr>
    <td class="tg-0lax">4</td>
    <td class="tg-0lax">5</td>
    <td class="tg-0lax">6</td>
  </tr>
</tbody>
</table>
````

**Reply by cscheid** [reply]
_style_ tags won't work in PDF format, but the table tags will. The documentation is [here](https://quarto.org/docs/prerelease/1.3/tables.html). Note that you can't just write the table directly like that. You need to wrap the code in a raw block, like so:

````
```{=html}
<table class="tg">
<thead>
  <tr>
    <th class="tg-0lax">C1</th>
    <th class="tg-0lax">C2</th>
    <th class="tg-0lax">C3</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td class="tg-tf2e">1</td>
    <td class="tg-tf2e">2</td>
    <td class="tg-tf2e">3</td>
  </tr>
  <tr>
    <td class="tg-0lax">4</td>
    <td class="tg-0lax">5</td>
    <td class="tg-0lax">6</td>
  </tr>
</tbody>
</table>
```
````

**Comment by plierouge** [comment]
![pdf](https://github.com/quarto-dev/quarto-cli/assets/136630150/c8f2ffaa-0c5a-4578-8d40-6f87afb64c71)

**Reply by mcanouil** [reply, marked as answer]
No worries, it's fine.

I apologise as I realise the documentation (there is an ongoing PR to improve this exact part) is implying some components, so here is a full example on how to use HTML tables to be used in PDF.

<table>
<tr><th>Quarto document</th><th>PDF</th></tr>
<tr>
<td>

````qmd
---
title: "Quarto Playground"
format: pdf
bibliography: ref.bib
---

```{=html}
<table>
  <caption>
    <span data-qmd="As described in @einstein1905, computers are great.">
    </span>
  </caption>
  <thead>
    <tr>
      <th><span data-qmd="_Header 1_"></span></th>
      <th><span data-qmd="_Header 2_"></span></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><span data-qmd="__Bold Markdown text__"></span></td>
      <td>Regular output</td>
    </tr>
  </tbody>
</table>
```
````

</td>
<td>
<img width="578" alt="screenshot of the rendered PDF from the code above which includes a title, a caption with a reference, a small two by two table with italics headers, and below the full reference." src="https://github.com/quarto-dev/quarto-cli/assets/8896044/babd67e3-7744-4d79-9a64-6534f408a53d">
</td>
</tr>
</table>

**Reply by plierouge** [reply]
Ah... fantastic!

**Reply by plierouge** [reply]
Another question... 
Is it possible to decide where  to put the horizontal lines in the pdf table output? Typically, I would like to have a "footnote table" under the last line of the table. See my amino acid table in the pdf output for an example.

Thanks again for your patience and your great advice!

**Reply by mcanouil** [reply]
I have to admit that I don't know for sure, but the HTML tag `<tfoot></tfoot>` might work.

**Reply by mcanouil** [reply]
@plierouge if you feel your question has been addressed, could you mark the answer?

**Reply by plierouge** [reply]
The tfoot tag worked, but the pdf output has a line under it. But I am getting close. I will keep searching! Thanks!
